torch/csrc/distributed/rpc/process_group_agent.cpp

```diff
@@ -673,6 +673,15 @@ std::unordered_map<std::string, std::string> ProcessGroupAgent::getMetrics() {
   return metrics;
 }
 
+TypeResolver ProcessGroupAgent::getTypeResolver() {
+  auto typeResolver = [&](const std::string& typeStr) {
```

Comment:
any reason for creating a new lambda every time calling this? This will be on the critical path as we need this for every message. Is it possible to have a static var here, and share it through `shared_ptr` or reference?